compiler/src/iree/compiler/Codegen/Common/GenericVectorization.cpp

```diff
@@ -193,14 +199,16 @@ void GenericVectorizationPass::runOnOperation() {
   });
   for (auto op : candidates) {
     SmallVector<int64_t> vectorSizes;
+    SmallVector<bool> scalableVecDims;
```

Comment:
In practice these vectors will be fairly small and need to be converted to `ArrayRef<bool>/SmallVector<bool>` for the MLIR APIs anyway, so I'm unsure this would be of much benefit.

Reply:
We dont need to convert it. We can always pass around `BitVector`s.

Reply:
They would need to be converted as these need to be passed to the linalg vectorizer, which expects an `ArrayRef<bool>`. They there would also need to converted to BitVectors from the lowering config (which stores them as bools, and I'm not sure there's a way to represent a BitVector directly there).